portal-impl/src/com/liferay/portal/security/permission/ResourceActionsImpl.java

```diff
@@ -609,6 +609,16 @@ public void removeModelResource(String name, String action) {
 		}
 	}
 
+	@Override
+	public void removeResourceReference(Portlet portlet, String resourceName) {
+		_resourceReferences.remove(resourceName);
+
+		if (portlet != null) {
+			_portletRootModelResources.remove(portlet.getPortletId());
+			_resourceReferences.remove(portlet.getPortletName());
+		}
```

Comment:
Not sure if is correct. @carolmariaabb could you help here?

Reply:
@devpedrotavares you will need infra-core team review for this change 😞

Reply:
@devpedrotavares Please forward this PR to the `liferay-core-infra` user. Be sure to add detailed steps on how to reproduce the issue and more context such as why we need to create this new method. It may be difficult for them to accept this change.

@murilostodolni could you help Pedro with this? 

Thanks!